products/knownprojects/python/extractors.py

```diff
@@ -8,81 +8,69 @@
 from shapely import wkb
 
 from . import RAW_DATA_PATH
-from .utils import ETL, hash_each_row
+from .utils import ETL
 
 
 @ETL
-def dcp_knownprojects() -> pd.DataFrame:
-    filename = "kpdb_20211006_shapefiles.zip"
+def dcp_knownprojects(filename: str) -> pd.DataFrame:
     df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
     df.rename(columns={"geom": "geometry"}, inplace=True)
     df["geometry"] = df["geometry"].fillna()
     return df
 
 
 @ETL
-def esd_projects() -> pd.DataFrame:
-    filename = "2021.2.10 State Developments for Housing Pipeline.xlsx"
+def esd_projects(filename: str) -> pd.DataFrame:
     df = pd.read_excel(f"{RAW_DATA_PATH}/{filename}", dtype=str)
     return df
 
 
 @ETL
-def edc_projects() -> pd.DataFrame:
-    filename = "2022.11.18 EDC inputs for DCP housing projections.xlsx"
+def edc_projects(filename: str) -> pd.DataFrame:
     df = pd.read_excel(f"{RAW_DATA_PATH}/{filename}", dtype=str)
     return df
 
 
 @ETL
-def dcp_n_study() -> pd.DataFrame:
-    filename = "nstudy_rezoning_commitments_shapefile_20221017.zip"
+def edc_dcp_inputs(filename: str) -> gpd.GeoDataFrame:
     df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
     return df
 
 
 @ETL
-def dcp_n_study_future() -> pd.DataFrame:
-    filename = "future_nstudy_shapefile_20221017.zip"
+def dcp_n_study(filename: str) -> pd.DataFrame:
     df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
     return df
 
 
 @ETL
-def dcp_n_study_projected() -> gpd.geodataframe.GeoDataFrame:
-    filename = "PastNeighborhoodStudies_20221019.zip"
+def dcp_n_study_future(filename: str) -> pd.DataFrame:
     df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
     return df
 
 
 @ETL
-def hpd_rfp() -> pd.DataFrame:
-    filename = "20221122_HPD_RFPs.xlsx"
-    df = pd.read_excel(f"{RAW_DATA_PATH}/{filename}", dtype=str)
+def dcp_n_study_projected(filename: str) -> gpd.GeoDataFrame:
+    df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
     return df
 
 
 @ETL
-def hpd_pc() -> pd.DataFrame:
-    filename = "2022_11_23 DCP_SCA Pipeline.xlsx"
+def hpd_rfp(filename: str) -> pd.DataFrame:
     df = pd.read_excel(f"{RAW_DATA_PATH}/{filename}", dtype=str)
     return df
 
 
 @ETL
-def dcp_planneradded():
-    filename = "dcp_planneradded_2023_03_21.csv"
-    df = pd.read_csv(f"{RAW_DATA_PATH}/{filename}", dtype=str)
-    df.rename(columns={"WKT": "geometry"}, inplace=True)
-    # print(df)
+def hpd_pc(filename: str) -> pd.DataFrame:
+    df = pd.read_excel(f"{RAW_DATA_PATH}/{filename}", dtype=str)
     return df
 
 
 @ETL
-def edc_dcp_inputs() -> gpd.geodataframe.GeoDataFrame:
-    filename = "edc_shapefile_20221118.zip"
-    df = gpd.read_file(f"zip://{RAW_DATA_PATH}/{filename}")
-    # print(df)
+def dcp_planneradded(filename: str) -> pd.DataFrame:
+    df = pd.read_csv(f"{RAW_DATA_PATH}/{filename}", dtype=str)
+    df.rename(columns={"WKT": "geometry"}, inplace=True)
     return df
```

Comment:
It seems like the extractors can really just be abstracted out to one or two functions at this point, no? Given that filename is now an arg, this etl isn't exactly specific to "dcp_planneradded" - it's just pd.read_csv with a rename (which I believe could even be done with a kwarg to read_csv)

Reply:
maybe. there seems to be a little bit of source-specific stuff like [here](https://github.com/NYCPlanning/data-engineering/blob/f665ddca6e560cc9c87cf004eae3b43a9ce2f655/products/knownprojects/python/extractors.py#L18). that line could be done to all source data, but I was aiming for minimal unrelated change

and I think it'd have to be 3 functions to cover the 3 file types we get from the Housing team ([list of filenames](https://github.com/NYCPlanning/data-engineering/blob/f665ddca6e560cc9c87cf004eae3b43a9ce2f655/products/knownprojects/python/__init__.py#L25)). or could be one function with branching logic based on the filename's extension?